### PR TITLE
Standardize on the way we wait for InputSystem validity.

### PR DESCRIPTION
**Pull request by wiwei**
There are multiple ways we wait for InputSystem validity, some of which
check for the current state of the object and some of which use the await directly.

AFAICT it looks like we should have moved to a common helper (i.e. in this PR: https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3462/files), so this common-izes this action into the base class

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalListener.cs [hunk 1]
```diff
@@ -13,6 +14,10 @@ public class InputSystemGlobalListener : MonoBehaviour
     {
         private bool lateInitialize = true;
 
+        /// <summary>
+        /// Prefer using EnsureInputSystemValid(), which will only await if the input system isn't
+        /// yet valid.
+        /// </summary>
         protected readonly WaitUntil WaitUntilInputSystemValid = new WaitUntil(() => MixedRealityToolkit.InputSystem != null);
```
Comment (luis-valverde-ms) [inline Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalListener.cs]: Looks good. Should we mark this one as deprecated so users don't take dependencies on it?

Comment (wiwei) [reply Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalListener.cs]: Done

**Comment by wiwei** [comment]
I'm waiting on the reporter to verify the fix.

**Comment by wiwei** [comment]
Reporter verified it, thanks!

**Pull request merged**